helix-view/src/document.rs

```diff
@@ -763,12 +768,12 @@ impl Document {
 
         if !transaction.changes().is_empty() {
             self.version += 1;
+            let reverted_tx = transaction.invert(&old_doc);
```

Comment:
Move this into the `if`

Reply:
I overlooked this when switching from a change set based approach to an async based approach. I will revert those hunks back to master. Apologies! I am on the go right now will update later

Reply:
I have removed  this change from the PR